snake/snake_game_khr.py

```diff
@@ -0,0 +1,150 @@
+                new_board.append(row)
+            elif i in {self.height - 1}:
+                row = [self.bottom_boarders] * self.width
+                new_board.append(row)
+            else:
+                row = []
+                for j in range(self.width):
+                    if j in {0, self.width - 1}:
+                        row.append(self.vert_boarders)
+                    else:
+                        row.append(" ")
+                new_board.append(row)
+        return new_board
+
+    def show(self):
+        for row in self.board:
+            print(' '.join(row))
+
+
+class Snake:
+    def __init__(self, symbol="o", position=(1, 1)):
+        self.symbol = symbol
+        self.body = [position]
+
+    def eat(self, position: tuple):
+        self.body.append(position)
+
+    def move(self, position: tuple):
+        self.body.append(position)
+        self.body.pop(0)
+
+
+class Apple:
+    def __init__(self, symbol='$', position=(2, 2)):
+        self.position = [position]
+        self.symbol = symbol
+
+
+class Game:
+    def __init__(self):
+        self.symbol = "o"
+        self.width = 20
+        self.height = 10
+        self.board = Board(self.width, self.height)
+        self.snake = Snake()
+        self.apple = Apple()
+
+    def init_apple(self):
+        while True:
+            new_apple_loc = (
+                randint(1, self.width - 1),
+                randint(1, self.height - 1),
+            )
+            if new_apple_loc not in self.snake.body:
+                break
+        self.apple = Apple(position=new_apple_loc)
+
+    def play(self):
+        new_pos = (1, 2)
+        self.snake.eat(new_pos)
+        self.render()
+        sleep(2)
+
+        self.snake.move((1, 3))
+        self.render()
+        sleep(2)
+
+        # apple = (2, 4)
+        self.snake.move((1, 4))
+        self.render()
+        sleep(2)
+
+        self.snake.move((2, 4))
+        self.render()
+        sleep(2)
+
+        self.snake.move((2, 3))
+        self.render()
+        sleep(2)
+        apple = (2, 2)
+
+        self.snake.eat(apple)
+        self.render()
+        sleep(2)
+
+        self.init_apple()
+        self.render()
+        sleep(2)
+
+        self.snake.move((2, 1))
+        self.render()
+        sleep(2)
+
+        self.snake.move((3, 1))
+        self.render()
+        sleep(2)
+
+    def clear(self):
+        self.board = Board(self.width, self.height)
+        if os.name == 'posix':
+            os.system('clear')
+        else:
+            os.system('cls')
+
+    def render(self):
+        self.clear()
+        for i, j in self.apple.position:
+            self.board.board[i][j] = self.apple.symbol
+        for i, j in self.snake.body:
+            self.board.board[i][j] = self.snake.symbol
+            for x, y in self.apple.position:
+                if self.snake.body[0] == self.apple.position:
+                    self.snake.body.insert(0, self.apple.position)
```

Comment:
I'm not following why we check if the snake tail is equal apple, and then we make apple position as new snake tail.. You can just append apple to snake when you eat it, head + 1